app/ts/components/subcomponents/ConfigureRpcConnection.tsx

```diff
@@ -0,0 +1,266 @@
+import { createContext, type ComponentChildren } from 'preact'
+import { useComputed, useSignal, useSignalEffect } from '@preact/signals'
+import { useContext, useRef } from 'preact/hooks'
+import { Network, JsonRpcProvider } from 'ethers'
+import { AsyncStates, useAsyncState } from '../../utils/preact-utilities.js'
+import { TextInput } from './TextField.js'
+import { RpcEntries, RpcEntry } from '../../types/rpc.js'
+import { CHAIN_NAMES } from '../../utils/chainNames.js'
+import { sendPopupMessageToBackgroundPage } from '../../background/backgroundUtils.js'
+import { getSettings } from '../../background/settings.js'
+
+type ConfigureRpcContext = {
+	queryRpcInfo: (url:string) => void
+	rpcQuery: ReturnType<typeof useAsyncState<Network>>['value']
+	resetRpcQuery: () => void
+}
+
+const ConfigureRpcContext = createContext<ConfigureRpcContext | undefined>(undefined)
+
+const RpcQueryProvider = ({ children }: { children: ComponentChildren }) => {
+	const { value: rpcQuery, waitFor, reset: resetRpcQuery } = useAsyncState<Network>()
+
+	const queryRpcInfo = (url: string) => waitFor(async () => {
+		const provider = new JsonRpcProvider(url)
+		return await provider.getNetwork()
+	})
+
+	return <ConfigureRpcContext.Provider value = { { queryRpcInfo, rpcQuery, resetRpcQuery } }>{ children }</ConfigureRpcContext.Provider>
+}
+
+function useQueryRpc() {
+	const context = useContext(ConfigureRpcContext)
+	if (!context) throw new Error('useQueryRpc can only be used within children of RpcQueryProvider')
+	return context
+}
+
+export const ConfigureRpcConnection = ({ rpcEntries, rpcInfo }: { rpcEntries: RpcEntries, rpcInfo?: RpcEntry | undefined }) => {
+	const modalRef = useRef<HTMLDialogElement>(null)
+
+	const showConfigurationModal = () => modalRef.current?.showModal()
+
+	const cancelAndCloseModal = () => modalRef.current?.close()
+
+	const saveRpcEntry = (rpcEntry: RpcEntry) => {
+		sendPopupMessageToBackgroundPage({
+			method: 'popup_set_rpc_list',
+			data: [rpcEntry].concat(rpcEntries.filter(entry => entry.httpsRpc !== rpcEntry.httpsRpc))
+		})
+	}
+
+	const removeRpcEntryByUrl = async (url: string) => {
+		const { currentRpcNetwork } = await getSettings()
+
+		const reducedRpcEntries = rpcEntries.filter(entry => entry.httpsRpc !== url)
+
+		await sendPopupMessageToBackgroundPage({ method: 'popup_set_rpc_list', data: reducedRpcEntries })
+
+		// switch rpc when the active one is being removed
```

Comment:
this issue also occurs if the active rpc is being modified. Eg, its rpc url is changed, the RPC URL is not going to be switched to that automatically as its cached.